docs/eeps/draft-modern-java.md

```diff
@@ -0,0 +1,110 @@
+
+On the other hand, we do not expect problems with plugins and many other dependency libraries behind `embulk-deps`, even if the top-level Embulk core is loaded as a "named module". Embulk plugins are loaded by `PluginClassLoader` and the Embulk core library dependencies are loaded by `EmbulkDependencyClassLoader`. Both class loaders extend [`URLClassLoader`](https://docs.oracle.com/javase/8/docs/api/java/net/URLClassLoader.html). See [Java Language Specification: Chapter 7. Packages and Modules](https://docs.oracle.com/javase/specs/jls/se11/html/jls-7.html), [EEP-7: Core Library Dependencies](./eep-0007.md) and [EEP-9: Class Loading, and Executable JAR File Structure](./eep-0009.md) for more details.
+
+Even in Java 9+ with the module system, `URLClassLoader` loads JAR files like in the class path (as "unnamed modules"), not like in the module path (as "named modules" or "automatic modules"). Both plugins and the core library dependencies will still be loaded like in the class path (as "unnamed modules") even if the top-level Embulk core is loaded in the module path as a "named module" or an "automatic module" in Java 9+.
+
+In light of the foregoing, we will consider loading the Embulk core as a "named module" when the following conditions are satisfied.
+
+* After Embulk removes `msgpack-core` from `embulk-spi`.
+* After Embulk unsupports Java 8.
+* After Embulk makes some decision about the logging driver.
+
+In other words, this will happen after Embulk v2.0 at the earliest. See also [EEP-8: Milestones to Embulk v1.0, and versioning strategies to follow](./eep-0008.md).
+
+However, even before the transition happens and the JAR files start having `module-info.class`, we may anyway want to start adding [the `Automatic-Module-Name` manifest attribute earlier in `META-INF/MANIFEST.MF`](https://docs.oracle.com/en/java/javase/11/docs/specs/jar/jar.html#modular-jar-files) to every JAR file, including `embulk-spi`, `embulk-core`, `embulk-deps`, many other `embulk-util-*` libraries, and plugins. At the same time, we'll have to take care of the merge of `embulk-spi` into `embulk-core` when adding the `Automatic-Module-Name` attribute.
```

Comment:
> we may anyway want to start adding [the `Automatic-Module-Name` manifest attribute earlier in `META-INF/MANIFEST.MF`](https://docs.oracle.com/en/java/javase/11/docs/specs/jar/jar.html#modular-jar-files) to every JAR file, including `embulk-spi`, `embulk-core`, `embulk-deps`, many other `embulk-util-*` libraries, and plugins.

Will this part work before removing `msgpack-core` from `embulk-spi`?

It means

* Adding Automatic-Module-Name
* After Embulk removes `msgpack-core` from `embulk-spi`.
* After Embulk unsupports Java 8.
* After Embulk makes some decision about the logging driver.

Reply:
Sorry, I didn't get the point of your question... At least I meant that "adding Automatic-Module-Name" would happen before removing `msgpack-core`, as written as "start adding the `Automatic-Module-Name` manifest attribute **earlier**".